crates/telio-dns/tests/nameserver.rs

```diff
@@ -217,17 +245,85 @@ impl WGClient {
         buffer
     }
 
+    fn build_ipv6_dns_request(dns_query: &[u8], test_type: DnsTestType) -> Vec<u8> {
+        let length = UDP_HEADER + dns_query.len();
+        let total_length = IPV6_HEADER + length;
+        let mut buffer = vec![0u8; MAX_PACKET];
+
+        let local_address = Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1);
+
+        {
+            let mut ip_response =
+                MutableIpv6Packet::new(&mut buffer).expect("Failed to create MutableIpv6Packet");
+            ip_response.set_version(6);
+            ip_response.set_traffic_class(0);
+            ip_response.set_flow_label(0);
+            ip_response.set_payload_length(length as u16);
+            ip_response.set_next_header(IpNextHeaderProtocols::Udp);
+            ip_response.set_hop_limit(255);
+            ip_response.set_source(local_address);
+            ip_response.set_destination(local_address);
+        }
+
+        {
+            let mut udp_response = MutableUdpPacket::new(&mut buffer[IPV6_HEADER..total_length])
+                .expect("Failed to create MutableUdpPacket");
+            udp_response.set_source(100);
+            if matches!(test_type, DnsTestType::BadUdpPortIpv6) {
+                udp_response.set_destination(54);
+            } else {
+                udp_response.set_destination(53);
+            }
+            udp_response.set_length(length as u16);
+            udp_response.set_payload(dns_query);
+            if matches!(test_type, DnsTestType::BadUdpChecksumIpv6) {
+                udp_response.set_checksum(0);
+            } else {
+                udp_response.set_checksum(ipv6_checksum(
+                    &udp_response.to_immutable(),
+                    &local_address,
+                    &local_address,
+                ));
+            }
+        }
+
+        buffer.truncate(total_length);
+        buffer
+    }
+
     fn client_address(&self) -> SocketAddr {
         self.client_address
     }
 }
 
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Eq, PartialEq)]
 enum DnsTestType {
```

Comment:
If we're adding a `Ipv6` suffix to the IPv6 lines then I think we should add a `Ipv4` suffix to the other lines